flexmeasures/api/v3_0/users.py

```diff
@@ -91,26 +102,82 @@ def index(self, account: Account, include_inactive: bool = False):
         :status 403: INVALID_SENDER
         :status 422: UNPROCESSABLE_ENTITY
         """
+        if (page is None and per_page is not None) or (
+            page is not None and per_page is None
+        ):
+            return {"message": "Both page and per_page should be passed"}, 400
+
+        if page and per_page:
+            if page < 1:
+                return {"message": "Page number should be greater than 0"}, 400
+            if per_page < 1:
+                return {"message": "Per page number should be greater than 0"}, 400
 
         if account is not None:
             check_access(account, "read")
             accounts = [account]
         else:
-            accounts = []
-            for account in db.session.scalars(select(Account)).all():
-                try:
-                    check_access(account, "read")
-                    accounts.append(account)
-                except (Forbidden, Unauthorized):
-                    pass
-
-        users = []
-        for account in accounts:
-            users += get_users(
-                account_name=account.name,
-                only_active=not include_inactive,
-            )
-        return users_schema.dump(users), 200
+            accounts = get_accessible_accounts()
+
+        if page and per_page:
+            all_users: list = [
```

Comment:
If we only need `all_users` to count the maximal number, then this is a slow way (whole objects being loaded). You can be inspired by the asset API:

```
filter_statement = GenericAsset.account_id.in_([a.id for a in accounts])
num_records = db.session.scalar(
    select(func.count(GenericAsset.id)).where(filter_statement)
)
```